src/Microsoft.OpenApi/Writers/OpenApiWriterAnyExtensions.cs

```diff
@@ -109,72 +109,31 @@ private static void WriteObject(this IOpenApiWriter writer, JsonObject entity)
-            {
-                if (jsonStrValue.TryGetValue<DateTimeOffset>(out var dto))
-                {
-                    writer.WriteValue(dto);
-                }
-                else if (jsonStrValue.TryGetValue<DateTime>(out var dt))
-                {
-                    writer.WriteValue(dt);
-                }
-                else if (jsonStrValue.TryGetValue<string>(out var strValue))
-                {
-                    // check whether string is actual string or date time object
-                    if (DateTimeOffset.TryParse(strValue, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var dateTimeOffset))
-                    {
-                        writer.WriteValue(dateTimeOffset);
-                    }
-                    else if (DateTime.TryParse(strValue, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var dateTime))
-                    { // order matters, DTO needs to be checked first!!!
-                        writer.WriteValue(dateTime);
-                    }
-                    else
-                    {
-                        writer.WriteValue(strValue);
-                    }
-                }
-            }
-
-            else if (valueKind == JsonValueKind.Number && primitive is JsonValue jsonValue)
-            {
-
-                if (jsonValue.TryGetValue<decimal>(out var decimalValue))
-                {
-                    writer.WriteValue(decimalValue);
-                }
-                else if (jsonValue.TryGetValue<double>(out var doubleValue))
-                {
-                    writer.WriteValue(doubleValue);
-                }
-                else if (jsonValue.TryGetValue<float>(out var floatValue))
-                {
-                    writer.WriteValue(floatValue);
-                }
-                else if (jsonValue.TryGetValue<long>(out var longValue))
-                {
-                    writer.WriteValue(longValue);
-                }
-                else if (jsonValue.TryGetValue<int>(out var intValue))
-                {
-                    writer.WriteValue(intValue);
-                }
-            }
-            else if (valueKind is JsonValueKind.False)
-            {
-                writer.WriteValue(false);
-            }
-            else if (valueKind is JsonValueKind.True)
-            {
-                writer.WriteValue(true);
-            }
+            if (jsonValue.TryGetValue(out string stringValue))
+                writer.WriteValue(stringValue);
+            else if (jsonValue.TryGetValue(out DateTime dateTimeValue))
+                writer.WriteValue(dateTimeValue.ToString("o", CultureInfo.InvariantCulture)); // ISO 8601 format
+            else if (jsonValue.TryGetValue(out DateTimeOffset dateTimeOffsetValue))
+                writer.WriteValue(dateTimeOffsetValue.ToString("o", CultureInfo.InvariantCulture));
+#if NET6_0_OR_GREATER
+            else if (jsonValue.TryGetValue(out DateOnly dateOnlyValue))
```

Comment:
and time only please :)

Reply:
Oh I didn't think those were two separate types. I understood DateOnly 😄